clothcalc_placeholder.js

```diff
@@ -7915,48 +7915,44 @@
                     }
                 };
                 _self.addTabOnMarketWindow = function (name, shortname, callback) {
-                    var first = false;
-                    var current = {};
-                    if (MarketWindow.twdb_showTab)
-                        current = {fn: MarketWindow.twdb_showTab, name: "MarketWindow.twdb_showTab"};
-                    else
-                        current = {fn: MarketWindow.showTab, name: "MarketWindow.showTab"};
-                    if (typeof save.MarketWindowOpen == "undefined") {
-                        first = true;
-                        save.MarketWindowOpen = MarketWindow.open.toString();
-                        save.MarketWindowTab = current.fn.toString();
-                    }
+                    var tabclick = function (win, id) {
+                        if (!MarketWindow.window)
+                            return;
+                        MarketWindow.window.activateTab(id).$('div.tw2gui_window_content_pane > *', MarketWindow.DOM).each(function (i, e) {
+                            if ($(e).hasClass('marketplace-' + id)){
+                                $(e).children().fadeIn();
+                                $(e).show();
+                            } else {
+                                $(e).children().fadeOut();
+                                $(e).hide();
+                            }
+                        });
+                        MarketWindow['TWDB-' + shortname]();
+                    };
                     try {
-                        var inject = "MarketWindow.window.addTab('" + name + "', '" + shortname + "', tabclick).appendToContentPane($('<div class=\"marketplace-" + shortname + "\"/>'));";
-                        var newfunction = MarketWindow.open.toString().replace(/MarketWindow.DOM/,inject + "MarketWindow.DOM");
-                        eval("(function ($) {" + "MarketWindow.open = " + newfunction + "})(jQuery);")
```

Comment:
Wow, this code was so ugly it should be criminal :D Thanks for getting rid of it.